### ISSUE TITLE
Inconsistent implementation on SWA

### ISSUE THREAD

**Issue by LuoXin-s** [pytorch/pytorch#74022]
### 🐛 Describe the bug

Why `swa_utils` implement SWA different from the original paper? For example, there is no way to do a cyclical lr schedule to average model, the API only supports annealing the lr from its initial value to a given lr in n epochs, after this, the lr is fixed, different with one may expect.

### Versions

1.10.0

**Comment by albanD** [comment]
cc @izmailovpavel who originally authored this.

**Comment by izmailovpavel** [comment]
Hi @LuoXin-s @albanD, for the native PyTorch implementation we decided to implemented the general tools needed to use SWA: weight averaging and the default schedule. 

https://github.com/izmailovpavel/torch_swa_examples -- examples reproducing results from the original paper using the torch implementation. We found that in most standard applications the constant schedule is sufficient.

You should be able to implement the cyclical schedule using the PyTorch [`torch.optim.lr_scheduler.CyclicLR`](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CyclicLR.html#torch.optim.lr_scheduler.CyclicLR) scheduler. Please let me know if you run into issues implementing your custom lr schedule for SWA.

**Comment by LuoXin-s** [comment]
@izmailovpavel Thank you for your reply. Yeah, I can implement it by myself, it is not too hard, but the provided API and documentation may confuse people like me. For example, after reading your paper, I going to find some implementation, and there is PyTorch official implementation, that is cool, but in official docs, we can't find any instructions for how to implement SWA based cyclic scheduler. Also the paper "SWA object detection" indicated that the cyclic scheduler is more performant than the constant lr scheduler, so the people who come from the object detection community may expect there is official instruction in PyTorch docs.

**Comment by LuoXin-s** [comment]
@izmailovpavel Yeah, I think I can do that, I'll make a PR later.

**Comment by albanD** [comment]
Yes updating the doc to avoid confusion sounds like the right way to go here!
Happy to handle the landing of the PR once you've reviewed it!
